File: 1553-count-triplets-that-can-form-two-arrays-of-equal-xor/count-triplets-that-can-form-two-arrays-of-equal-xor.cpp
class Solution {
public:
    int countTriplets(vector<int>& arr) {
        vector<int> prefixSum;
        prefixSum.push_back(0); // start point
        int a = 0; 
        for (int i : arr) {
            a ^= i; 
            prefixSum.push_back(a); 
        }

        int res = 0; 
        for (int i = 0; i < prefixSum.size(); ++i) {
            for (int j = i + 1; j < prefixSum.size(); ++j) {
                if (prefixSum[i] == prefixSum[j]) {
                    res += (j - (i + 1)); 
                }
            }
        }
        return res; 
    }
};

File: 1553-count-triplets-that-can-form-two-arrays-of-equal-xor/count-triplets-that-can-form-two-arrays-of-equal-xor.py
class Solution:
    def countTriplets(self, arr: List[int]) -> int:
        prefix_sum = [0,]
        ps = 0
        for num in arr: 
            ps ^= num
            prefix_sum.append(ps)

        n = len(prefix_sum)
        res = 0
        for i in range(n): # i + 1: represents the start point
            for j in range(i + 1, n):
                # this means the XOR sum from i + 1 -> j = 0
                if (prefix_sum[i] == prefix_sum[j]): 
                    # count the number of valid triplets
                    res += (j - (i + 1))
        return res